gatsby/content/blog/2022/05/2022-05-18-synapse-1.59.mdx

```diff
@@ -0,0 +1,82 @@
+introduction of a generic worker application type.
+
+Synapse 1.59 furthers this work by deprecating the `synapse.app.appservice` and
+`synapse.app.user_dir` worker application types. Homeserver administrators
+should change the configuration of instances using these types to the generic
+`synapse.app.generic_worker` application type, and use the
+`notify_appservices_from_worker` and `update_user_directory_from_worker` to
+delegate application service and user directory work (respectively) to a worker.
+
+See the [upgrade
+notes](https://matrix-org.github.io/synapse/v1.59/upgrade.html#deprecation-of-the-synapseappappservice-and-synapseappuser_dir-worker-application-types)
+for more information on this change.
+
+## Push rules
+
+Push rules allow Matrix clients to define notification rules on a homeserver.
+One often reported issue with notification in Matrix is the fact that
+notifications are sent out when server ACLs, which define which server(s) can
+and cannot interact in a room, change. This is especially annoying during big
+waves of abuse, as there might be multiple servers that need to be banned from
+rooms, thus causing a lot of unneeded notifications.
+
+Synapse 1.59 now silences server ACL updates, by implementing the new push rule
+documented in
+[MSC3786](https://github.com/matrix-org/matrix-spec-proposals/pull/3786).
+
+Synapse 1.59 also allows third-party modules to validate and change the actions
+associated with an existing push rule via the [Module
+API](https://github.com/matrix-org/synapse/blob/d24a1486e5c5d69a8798a9d159fd9e06dfc8c3e3/synapse/module_api/__init__.py#L1360-L1420).
+This is helpful for modules wishing to, for example, configuring specific
+notification settings for new users register.
+
+## Everything else
+
+As of Synapse 1.59, Synapse will not communicate the name of devices over
+federation (unless configured otherwise), in order to better preserve user
+privacy. See the [upgrade
+notes](https://matrix-org.github.io/synapse/v1.59/upgrade.html#device-name-lookup-over-federation-has-been-disabled-by-default)
+for more information.
+
+Also note that we have issued a patch release for 1.59 (1.59.1) which fixes a
+long-standing bug that started to bite a good amount of server administrators.
+Server admins that are looking into upgrading their instance to Synapse 1.59 are
+recommended to upgrade to 1.59.1 rather than 1.59.0.
+
+See the [full
+changelog](https://github.com/matrix-org/synapse/releases/tag/v1.59.0) for a
+complete list of changes in this release. Also please have a look at the
+[upgrade
+notes](https://matrix-org.github.io/synapse/v1.59/upgrade#upgrading-to-v1590)
+for this version.
+
+Synapse is a Free and Open Source Software project, and we'd like to extend our
+thanks to everyone who contributed to this release, including (in no particular
+order) [Beeper](https://www.beeper.com/), [Dirk
+Klimpel](https://github.com/dklimpel), [Šimon
+Brandner](https://github.com/SimonBrandner),
```

Comment:
Šimon works for Element, so I'm not sure if it's appropriate to shout out to him here... otoh no real reason why not I guess!

Reply:
The biggest chunk of the work was done before they joined so I didn't think it was fair to consider it all an Element contribution.